agents-for-bedrock/features-examples/09-fine-grained-access-permissions/003_bedrock-agent/agent/agent.py

```diff
@@ -0,0 +1,287 @@
+import json
+import boto3
+import os
+import logging
+import cognitojwt
+
+
+# Set up logging
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+verified_permissions_client = boto3.client("verifiedpermissions")
+
+
+def get_named_parameter(event, name):
+    return next(item for item in event['parameters'] if item['name'] == name)['value']
+
+
+def list_claims(event):
+    # TODO: Implement logic to retrieve and return a list of claims
+    dummy_claims = [
+        {
+            "id": 1,
+            "claimAmount": 1000.0,
+            "claimDescription": "Dummy claim 1",
+            "claimStatus": "approved"
+        },
+        {
+            "id": 2,
+            "claimAmount": 2500.75,
+            "claimDescription": "Dummy claim 2",
+            "claimStatus": "pending"
+        }
+    ]
+    return dummy_claims
+
+
+def get_claim(event):
+    claim_id = int(get_named_parameter(event, 'claimId'))
+    # TODO: Implement logic to retrieve and return a claim by ID
+    dummy_claim = {
+        "id": claim_id,
+        "claimAmount": 5000.0,
+        "claimDescription": "Dummy claim",
+        "claimStatus": "denied"
+    }
+    return dummy_claim
+
+
+def update_claim(event):
+    claim_id = int(get_named_parameter(event, 'claimId'))
+    # TODO: Implement logic to update and return a claim by ID
+    dummy_claim = {
+        "id": claim_id,
+        "claimAmount": 3000.0,
+        "claimDescription": "Updated dummy claim",
+        "claimStatus": "approved"
+    }
+    return dummy_claim
+
+
+def lambda_handler(event, context):
+    logger.info(f'event: {event}')
+    logger.info(f'context: {context}')
+
+    sessionAttributes = event.get("sessionAttributes")
```

Comment:
we should add a comment that here is where the sessionAttribute is recovered from the sessionState

Reply:
done